src/main/java/uk/gov/companieshouse/api/accounts/transformer/CurrentPeriodTransformer.java

```diff
@@ -17,12 +19,18 @@ public CurrentPeriodEntity transform(CurrentPeriod entity) {
         CurrentPeriodEntity currentPeriodEntity = new CurrentPeriodEntity();
         CurrentPeriodDataEntity currentPeriodDataEntity = new CurrentPeriodDataEntity();
         BalanceSheetEntity balanceSheetEntity = new BalanceSheetEntity();
+        FixedAssetsEntity fixedAssetsEntity = new FixedAssetsEntity();
 
         BeanUtils.copyProperties(entity, currentPeriodDataEntity);
         if (entity.getBalanceSheet() != null) {
             BeanUtils.copyProperties(entity.getBalanceSheet(), balanceSheetEntity);
+        
+            if(entity.getBalanceSheet().getFixedAssets() !=null ) {
+                BeanUtils.copyProperties(entity.getBalanceSheet().getFixedAssets(), fixedAssetsEntity);
```

Comment:
I would advise creating variables within the required scope (within the if statement). It makes the code easier to navigate as the transforming code grows larger and makes it easier to refactor it into methods later.
```java
if(entity.getBalanceSheet().getFixedAssets() != null ) {
    FixedAssetsEntity fixedAssetsEntity = new FixedAssetsEntity();
    BeanUtils.copyProperties(entity.getBalanceSheet().getFixedAssets(), fixedAssetsEntity);
    balanceSheetEntity.setFixedAssets(fixedAssetsEntity);
```

Reply:
I believe Sonar may have a complaint about that though. I agree to a point about readability, but we shouldn't instantiate variables if they're not required - it's more cost effective in the way its currently implemented.